### PR TITLE
SF-5163 | Added DateFormatter relative format for future dates

### PR DESCRIPTION
**Pull request by Alex-Murtha**

### CODE REVIEW

File: src/sdk/formatting/formatDate.ts [hunk 1]
```diff
@@ -99,6 +106,16 @@ const relativeTimeLabels = {
 	dd: '%d days ago'
 };
 
+const futureRelativeTimeLabels = {
```
Comment (smbecker) [inline src/sdk/formatting/formatDate.ts:109]: These do not follow our [data formatting guidelines](https://skysync.atlassian.net/wiki/spaces/PUCS/pages/323649654/Data+formats).

**Review by smbecker — CHANGES_REQUESTED** [review]

File: src/sdk/formatting/formatDate.ts (hunk 1 above)
Comment (Alex-Murtha) [reply src/sdk/formatting/formatDate.ts:109]: Would I only use Today, Tomorrow, or a date instead of these?

Comment (smbecker) [reply src/sdk/formatting/formatDate.ts:109]: The guidelines only call out distant past, not distant future.

File: src/sdk/formatting/formatDate.ts [hunk 2]
```diff
@@ -138,11 +138,17 @@ const getConditionalRelativeFormat = (value: number, threshold: number, labelOne
 };
 
 const getRelativeFormat = (value: RelativeDate, options: DateFormatOptions): FormatMethod => {
+	const isPast = value.isPast;
```
Comment (smbecker) [inline src/sdk/formatting/formatDate.ts:141]:
```suggestion
	if (value.isPast) {
		return null;
	}	
```

File: src/sdk/formatting/formatDate.ts [hunk 3]
```diff
@@ -138,11 +138,17 @@ const getConditionalRelativeFormat = (value: number, threshold: number, labelOne
...
-		|| (options.allowRelativeInDistantPast && getConditionalRelativeFormat(value.days, relativeTimeThresholds.days, relativeTimeLabels.d, relativeTimeLabels.dd, options));
+	
+	if (isPast) {
+		return getConditionalRelativeFormat(value.seconds, relativeTimeThresholds.seconds, null, relativeTimeLabels.ss)
+			|| (options.allowRelativeInDistantPast && getConditionalRelativeFormat(value.minutes, relativeTimeThresholds.minutes, relativeTimeLabels.m, relativeTimeLabels.mm))
+			|| (options.allowRelativeInDistantPast && getConditionalRelativeFormat(value.hours, relativeTimeThresholds.hours, relativeTimeLabels.h, relativeTimeLabels.hh))
+			|| (options.allowRelativeInDistantPast && getConditionalRelativeFormat(value.days, relativeTimeThresholds.days, relativeTimeLabels.d, relativeTimeLabels.dd, options));
+	} else {
+		return null;
+	}
```
Comment (smbecker) [inline src/sdk/formatting/formatDate.ts:151]:
```suggestion
	return getConditionalRelativeFormat(value.seconds, relativeTimeThresholds.seconds, null, relativeTimeLabels.ss)
			|| (options.allowRelativeInDistantPast && getConditionalRelativeFormat(value.minutes, relativeTimeThresholds.minutes, relativeTimeLabels.m, relativeTimeLabels.mm))
			|| (options.allowRelativeInDistantPast && getConditionalRelativeFormat(value.hours, relativeTimeThresholds.hours, relativeTimeLabels.h, relativeTimeLabels.hh))
			|| (options.allowRelativeInDistantPast && getConditionalRelativeFormat(value.days, relativeTimeThresholds.days, relativeTimeLabels.d, relativeTimeLabels.dd, options));
```

**Review by smbecker — CHANGES_REQUESTED** [review]

**Review by smbecker — APPROVED** [review]

**Pull request merged**